#!/bin/sh
#PBS -q batch
#PBS -l walltime=72:00:00
#PBS -m abe

# use script: qsub -d . -l nodes=1:ppn=14 -v reads="PE",stranded="yes",dataset="/projects/anczukow-lab/fastq_files/test_reads",readlength="100",outdir="/projects/anczukow-lab/yuriem/test_pipeline_parallel",threads="14",stargenome="/projects/anczukow-lab/reference_genomes/human/Gencode/star_overhangs/star_100",gtf="/projects/anczukow-lab/reference_genomes/human/Gencode/gencode.v32.annotation.gtf",mismatch="2" /data/yuriem/scripts/pipeline_splicing_with_arguments_parallel_part1.pbs


#mkdir ${outdir}/fastqc
mkdir ${outdir}/QC_raw
mkdir ${outdir}/trimmed
mkdir ${outdir}/star_mapped
#mkdir rMATS_out

if [ $reads == "PE" ]; then

	ls ${dataset}/*_R1*.f* | grep -v Undetermined > ${outdir}/list1.txt

else 
	ls ${dataset}/*.f* | grep -v Undetermined > ${outdir}/list1.txt

fi

THREADS=`wc -l < list1.txt`

qsub -d . -V -t 1-$THREADS -v reads=$reads,stranded=$stranded,dataset=$dataset,readlength=$readlength,outdir=$outdir,threads=$threads,stargenome=$stargenome,gtf=$gtf,mismatch=$mismatch /data/yuriem/scripts/run_pipeline_Olga_job1.sh

echo "Samples:" > ${outdir}/run_pipeline_Olga_job1_log.txt

cat ${outdir}/list1.txt >> ${outdir}/run_pipeline_Olga_job1_log.txt

echo "Parameters: reads=" $reads ",stranded=" $stranded ",dataset=" $dataset ",readlength=" $readlength ",outdir=" $outdir ",threads=" $threads ",stargenome=" $stargenome ",gtf=" $gtf ",mismatch=" $mismatch >> ${outdir}/run_pipeline_Olga_job1_log.txt

