#!/bin/sh
#PBS -q batch
#PBS -l nodes=1:ppn=14
#PBS -l walltime=24:00:00
#PBS -m abe

# use script: qsub -d . -v reads="PE",outdir="/projects/anczukow-lab/yuriem/test_pipeline",threads="14",gtf="/data/yuriem/bin/genomes/Annotation/Homo_sapiens.GRCh38.98.gtf" /data/yuriem/scripts/pipeline_splicing_with_arguments_parallel_part2.pbs

if [ $reads == "PE" ]; then

        grep -e "Started analysis of " -e "Both Surviving" run_pipeline_Olga_job1.sh.e* | grep -v '_R2*.f' | tr '\n' '\t' | sed 's/Started analysis/\nStarted analysis/g' | awk '{print $4 "\t" $8 "\t" $11 " " $12 "\t" $16 " " $17 "\t" $21 " " $22 "\t" $24 " " $25}' | sed '/^[[:space:]]*$/d' | awk 'NR==1{print "sample\t# of reads\tBoth surviving\tForward only surviving\t Reverse only surviving\t Dropped"}{print $0}' > ${outdir}/trimming_stats.txt

else
        grep -e "Started analysis of " -e "Surviving" run_pipeline_Olga_job1.sh.e* | tr '\n' '\t' | sed 's/Started analysis/\nStarted analysis/g' | awk '{print $4 "\t" $7 "\t" $9 " " $10 "\t" $12 " " $13}' | sed '/^[[:space:]]*$/d' | awk 'NR==1{print "sample\t# of reads\tBoth surviving\t\t Dropped"}{print $0}' > ${outdir}/trimming_stats.txt

fi

#Combine bam file for new gtf

sample=$(ls ${outdir}/star_mapped/*Aligned.sortedByCoord.out.bam | sed 's/Aligned.sortedByCoord.out.bam/.gtf/g')

ls ${sample} > assembly_gtf_list.txt

#Stringtie merge

/projects/anczukow-lab/tools/stringtie-2.0.6/stringtie --merge -G $gtf -o ${outdir}/star_mapped/stringtie_merged.gtf assembly_gtf_list.txt -p $threads


#rMATS

#module unload python
#module load python/2.7.13
#module load it-modules

#while read line
#do
#b1=$(echo $line | cut -d ' ' -f1) 
#b2=$(echo $line | cut -d ' ' -f2)
#sample=$(echo $line | cut -d ' ' -f3)
#read1=$(echo $b1 | sed 's/.txt/_gdc_realn_rehead_R1.fq/g' | sed 's!star_mapped/!trimmed/!g')
#readlength=$( head -2 $read1 | tail -1 | wc -c)
#python /opt/compsci/rMATS/4.0.1/rMATS-turbo-Linux-UCS2/rmats.py --b1 $b1 --b2 $b2 --gtf star_mapped/final.gtf --od rMATS_out/${sample} -t paired --nthread 14 --readLength $readlength 
#done < rMATS_pairs.txt

